pkg/imgpkg/cmd/copy.go

```diff
@@ -196,10 +196,25 @@ func (o *CopyOptions) writeImagesLockOutput(processedImages *ctlimgset.Processed
 		},
 	}
 
-	for _, img := range processedImages.All() {
-		imagesLock.Images = append(imagesLock.Images, lockconfig.ImageRef{
-			Image: img.DigestRef,
-		})
+	if o.LockInputFlags.LockFilePath != "" {
+		var err error
+		imagesLock, err = lockconfig.NewImagesLockFromPath(o.LockInputFlags.LockFilePath)
+		if err != nil {
+			return err
+		}
+		for i, image := range imagesLock.Images {
+			img, found := processedImages.FindByURL(ctlimgset.UnprocessedImageRef{DigestRef: image.Image})
```

Comment:
is this going to not work in cases when UnprocessedImageRef included a Tag? do we need to not index on the tag, or does it matter?

Reply:
If you provide an ImagesLock file there is no place to add a tag. So I believe it is safe to ignore the tag in this particular situation.